src/policies/BaseRewardDistributor.sol

```diff
@@ -0,0 +1,370 @@
+            if (weeklyMerkleRoots[week] != bytes32(0)) {
+                // Verify proof safely, skip if invalid or already claimed
+                if (_verifyProofSafe(user_, week, amount, proofs_[i])) {
+                    claimableAmount += amount;
+                }
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+
+        // Calculate equivalent vault shares
+        if (claimableAmount > 0) {
+            vaultShares = REWARD_TOKEN_VAULT.previewWithdraw(claimableAmount);
+        }
+    }
+
+    /// @notice Claim rewards for one or more weeks in a single transaction
+    ///
+    /// @param  weeks_          Array of week numbers to claim
+    /// @param  amounts_        Array of amounts for each week (must match Merkle leaves)
+    /// @param  proofs_         Array of Merkle proofs, one per week
+    /// @param  asVaultToken_   If true, claim as vault token; if false, unwrap to underlying token
+    function claim(
+        uint256[] calldata weeks_,
+        uint256[] calldata amounts_,
+        bytes32[][] calldata proofs_,
+        bool asVaultToken_
+    ) external virtual onlyEnabled {
+        _validateClaimArrays(weeks_, amounts_, proofs_);
+
+        uint256 totalAmount = _processClaims(msg.sender, weeks_, amounts_, proofs_);
+
+        _transferRewards(msg.sender, totalAmount, weeks_.length, asVaultToken_);
+    }
+
+    // ========== INTERNAL HELPERS ========== //
+
+    /// @notice Validate claim input arrays
+    ///
+    /// @param  weeks_          Array of week numbers to claim
+    /// @param  amounts_        Array of amounts for each week
+    /// @param  proofs_         Array of Merkle proofs, one per week
+    function _validateClaimArrays(
+        uint256[] calldata weeks_,
+        uint256[] calldata amounts_,
+        bytes32[][] calldata proofs_
+    ) internal pure {
+        if (weeks_.length == 0) revert RewardDistributor_NoWeeksSpecified();
+        if (weeks_.length != amounts_.length || weeks_.length != proofs_.length) {
+            revert RewardDistributor_ArrayLengthMismatch();
+        }
+    }
+
+    /// @notice Process claims and return total amount
+    ///
+    /// @param  user_           The user address claiming rewards
+    /// @param  weeks_          Array of week numbers to claim
+    /// @param  amounts_        Array of amounts for each week
+    /// @param  proofs_         Array of Merkle proofs, one per week
+    /// @return totalAmount     The total amount claimed across all weeks
+    function _processClaims(
+        address user_,
+        uint256[] calldata weeks_,
+        uint256[] calldata amounts_,
+        bytes32[][] calldata proofs_
+    ) internal returns (uint256 totalAmount) {
+        for (uint256 i = 0; i < weeks_.length; ) {
+            uint256 week = weeks_[i];
+            uint256 amount = amounts_[i];
+
+            // Verify merkle root is set for this week
+            if (weeklyMerkleRoots[week] == bytes32(0))
```

Comment:
If there are no rewards for a particular week (hypothetically):

- If the merkle root were not set, this would result in a revert
- If the merkle root were set, would it be non-zero? In which case this would continue